### DISCUSSION TITLE
change what disk convert uses while working

### DISCUSSION THREAD

**Discussion by nesss01** [ImageMagick/ImageMagick#2608] [Help]
I am using the convert command with -negate. The output file will be rather large. Even if I direct convert to create the output file on an external drive with plenty of space the program seems to use disk space on my internal boot drive while running and the disk runs out of space.

How can i change the drive convert is using while doing its stuff?

**Comment by snibgo** [comment]
I suppose you mean IM writes a temporary file, which you can see on your disk while the command is running.

You can define the environment variable MAGICK_TEMPORARY_PATH before running your command. See https://imagemagick.org/script/resources.php

Or you can do it within the command with "-define registry:temporary-path". See https://imagemagick.org/script/defines.php
